libs/designsystem/src/lib/components/action-sheet/action-sheet.component.ts

```diff
@@ -0,0 +1,119 @@
+import {
+  Component,
+  ElementRef,
+  EventEmitter,
+  HostBinding,
+  HostListener,
+  Input,
+  Output,
+} from '@angular/core';
+
+import { PlatformService } from '../../helpers/platform.service';
+import { OpenState } from '../../models/open-state';
+import { ModalController } from '../modal/services/modal.controller';
+
+import { ActionSheetConfig } from './config/action-sheet-config';
+import { ActionSheetItem } from './config/action-sheet-item';
+
+@Component({
+  selector: 'kirby-action-sheet',
+  templateUrl: './action-sheet.component.html',
+  styleUrls: ['./action-sheet.component.scss'],
+})
+export class ActionSheetComponent {
+  private state = OpenState.closed;
+  attentionLevel: '2' | '3' = '3';
+  static readonly OPEN_DELAY_IN_MS = 100;
+
+  constructor(
+    private platform: PlatformService,
+    private modalController: ModalController,
+    private elementRef: ElementRef<HTMLElement>
+  ) {}
+
+  @Input() cancelButtonText = 'Cancel';
+  @Input() disabled: boolean = false;
+  @Input() header: string;
+  @Input() subheader: string;
+  @Input() items: Array<ActionSheetItem>;
+  @Input() triggerIconName = 'more';
+  @Input() triggerText?: string;
+  @Output() itemSelect: EventEmitter<ActionSheetItem> = new EventEmitter<ActionSheetItem>();
+  @Input() tabindex = 0;
+
+  @HostBinding('attr.tabindex')
+  get _tabindex() {
+    return this.disabled ? -1 : this.tabindex;
+  }
+
+  @HostBinding('class.is-open')
+  get isOpen(): boolean {
```

Comment:
Consider also adding a `_isOpening` property with a `class.is-opening` HostBinding to allow for animating the transition from closed => open (see dropdown for an example)

Reply:
Yeah, I actually considered that, but the dropdown only uses `.is-opening` to add `display: block` - no transition, So behaviour-wise they are actually equivalent now